### ISSUE TITLE
Use dependency graph and toposort, not straitforward execution (or at least have an option)

### ISSUE THREAD

**Issue by Jabher** [microsoft/just#202]
It would be nice to have a function `taskDependency` like 
```javascript

task("clean", () => {
    await fs.remove()
})

task("build-js", () => {
    await taskDependency("clean")
    //...build...
})
task("build-css", () => {
    await taskDependency("clean")
    //...build...
})
task("build", parallel("build-js", "build-css"))
```

In a manner that "clean" will execute only once (for now I can only declare `series` that will cause the re-run of same task twice).
I understand that I can declare more complex logic with sequences, but I cannot make logic like this.
```
A --- B --- C
 \    \     \
  D --- E --- F
```

**Comment by Jabher** [comment]
If needed - I think I can do it. I like the approach of just, but I'm unsure about whether I should participate or this is intended "no-behavior" that is complicating stuff

**Comment by kenotron** [comment]
Since we took undertaker as our task orchestrator, we followed the trend coming from gulp 4. So this article is pretty relevant: https://fettblog.eu/gulp-4-parallel-and-series/

In a sufficiently large just.config.js, you'd probably find that having a mess of dependent tasks is quite hard to manage (we reached a certain point and started asking constantly why a certain task is triggered). So, yes you can design a dependency runner kind of API, but I wouldn't be adding it to the orchestrator at this point for the same reason as the gulp 4.
